cmake/Packaging.cmake

```
@@ -98,16 +101,90 @@ endmacro()
 #
 # Linux packages
 #
+
+macro(configure_linux_package_name)
+  ##Get Distro name for fall back
+  execute_process(
+    COMMAND bash "-c" "cat /etc/os-release | grep ^ID= | sed 's/ID=//g'"
+    OUTPUT_VARIABLE _DISTRO_NAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE "\"" "" DISTRO_NAME "${_DISTRO_NAME}")
+  message(STATUS "DISTRO NAME: ${DISTRO_NAME}")
+
+  execute_process(
+    COMMAND bash "-c"
+            "cat /etc/os-release | grep ^ID_LIKE= | sed 's/ID_LIKE=//g'"
+    OUTPUT_VARIABLE _DISTRO_LIKE
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE "\"" "" DISTRO_LIKE "${_DISTRO_LIKE}")
+  message(STATUS "DISTRO LIKE: ${DISTRO_LIKE}")
+
+  execute_process(
+    COMMAND
+      bash "-c"
+      "cat /etc/os-release | grep ^VERSION_CODENAME= | sed 's/VERSION_CODENAME=//g'"
+    OUTPUT_VARIABLE _DISTRO_CODENAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE "\"" "" DISTRO_CODENAME "${_DISTRO_CODENAME}")
+  message(STATUS "DISTRO CODENAME: ${DISTRO_CODENAME}")
+
+  execute_process(
+    COMMAND bash "-c"
+            "cat /etc/os-release | grep ^VERSION_ID= | sed 's/VERSION_ID=//g'"
+    OUTPUT_VARIABLE _DISTRO_VERSION_ID
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE "\"" "" DISTRO_VERSION_ID "${_DISTRO_VERSION_ID}")
+  message(STATUS "DISTRO ID: ${DISTRO_VERSION_ID}")
+
+  #CHECK IF RUNNING ON GITHUB FOR A RUN ATTEMPT TO USE IT
+  set(PACKAGE_BUILD_ATTEMPT $ENV{GITHUB_RUN_ATTEMPT})
+  if("$ENV{GITHUB_RUN_ATTEMPT}" STREQUAL "")
+    set(PACKAGE_BUILD_ATTEMPT 1)
+  endif()
+
+  #Check if its Debian Like
+  string(REGEX MATCH debian|buntu DEBTYPE "${DISTRO_LIKE}")
+  if((NOT ("${DEBTYPE}" STREQUAL "")) OR ("${DISTRO_NAME}" STREQUAL "debian"))
+    if("${CPACK_DEBIAN_PACKAGE_RELEASE}" STREQUAL "")
+      set(CPACK_DEBIAN_PACKAGE_RELEASE
+          "${PACKAGE_BUILD_ATTEMPT}~${DISTRO_CODENAME}")
+    endif()
+    set(CPACK_GENERATOR "DEB")
+    message(STATUS "DEB RELEASE INFO: ${CPACK_DEBIAN_PACKAGE_RELEASE}")
+  endif()
+
+  check_is_rpm()
+
+  if("${DISTRO_NAME}" STREQUAL "arch")
+      set(DISTRO_VERSION_ID "") #this has no meaning no arch for packages
+  endif()
+  #Determain the code name to be used
+  if(NOT "${DISTRO_VERSION_ID}" STREQUAL "")
+    set(CN_STRING "${DISTRO_VERSION_ID}-")
+  endif()
+
+  if(NOT "${DISTRO_CODENAME}" STREQUAL "")
+    set(CN_STRING "${DISTRO_CODENAME}-")
+  endif()
+  set(OS_STRING "${DISTRO_NAME}-${CN_STRING}${CMAKE_SYSTEM_PROCESSOR}")
+
+  message(STATUS "OS INFO: ${OS_STRING}")
+
+endmacro()
+
 macro(configure_linux_packaging)
 
   message(VERBOSE "Configuring Linux packaging")
 
-  set(CPACK_GENERATOR "DEB;RPM")
+  # set fallback name
+  configure_linux_package_name()
```

```suggestion
  # Set fallback name.
  configure_linux_package_name()
```

Maybe some context around this might be helpful. Why set a fallback name? What will replace it?